### PR TITLE
Add ajax helpers

### LINKED ISSUE #40: New Ajax helpers
Would be nice to have ajax helpers (ala jquery). We are actually _copypasting_ the Ajax component on every Ralix app, so it seems to be a useful abstraction.

Example:

- `ajax`(url, params = {}, options = {})
- `get` & `post` => shortcuts using `ajax` under the hood with predefined "verb"

Considerations:
- we can use `fetch` internally
- we should return promises? this will allow us something like `post('...').then((result) => { insertHTML('...') })`
- Automatically parse JSONs ??

### PR DESCRIPTION
**Pull request by jfelip937**
Closes #40

### CODE REVIEW

File: docs/CORE_API.md [hunk 1]
```diff
@@ -173,3 +173,18 @@ Returns the element which received the last fired event.
 - `currentEvent()`
 
 Returns the last fired event.
+
+
```
Comment (markets) [inline docs/CORE_API.md:177]: ✂️  extra white line

File: docs/CORE_API.md [hunk 2]
```diff
@@ -173,3 +173,18 @@ Returns the element which received the last fired event.
...
+## Ajax
```
Comment (markets) [inline docs/CORE_API.md:178]: we should mention this also in the README, in the "summary" section (which links to this doc): https://github.com/ralixjs/ralix#core-methods

File: src/core.js [hunk 3]
```diff
@@ -179,6 +179,26 @@ export default class Core {
       _setDataset(el, attribute)
   }
 
+  ajax(path, params = {}, options = {}) {
```
Comment (markets) [inline src/core.js:182]: how will look a call that requires options, but doesn't need params?

Comment (jfelip937) [reply src/core.js:182]: `ajax(path, {}, { credentials: 'include' })`

Comment (jfelip937) [reply src/core.js:182]: Maybe better add params to options?

Comment (markets) [reply src/core.js]: Not sure, now that we have another param, I think this method signature is too large. Maybe we can just use something like: `ajax(path, { params = {}, options = {} })` and all custom options go inside the `options` object. Another approach: keep params separated and make options to handle all fetch options + extras.

**TL;DR**
1) ajax(path, params, options, dataType)
2) ajax(path, { params, options })
3) ajax(path, params, options = {})

File: src/core.js [hunk 4]
```diff
@@ -179,6 +179,28 @@ export default class Core {
       _setDataset(el, attribute)
   }
 
+  async ajax(path, params = {}, options = {}, dataType = 'text') {
+    const defaults = { method: 'GET', credentials: 'include' }
+    options = Object.assign({}, defaults, options)
+
+    if (['POST', 'PATCH', 'PUT'].includes(options.method)) options = Object.assign({}, { body: JSON.stringify(params) }, options)
```
Comment (markets) [inline src/core.js:186]: I think the following style is much more readable:

```
if (condition)
  a
else
  b
```

**Comment by markets** [comment]
@jfelip937 a couple of details more:
- dataType => format more ~Rails
- update the modal example in the README
